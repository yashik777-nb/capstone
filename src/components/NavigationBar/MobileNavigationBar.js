import React from "react";
import { IconButton } from "@material-ui/core";
import { Menu } from "@material-ui/icons";

class MobileNavigationBar extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      right: false,
    };
  }

  toggleDrawer = (anchor, open) => (event) => {
    if (
      event.type === "keydown" &&
      (event.key === "Tab" || event.key === "Shift")
    )
      return;

    this.setState({ [anchor]: open });
  };

  render() {
    return (
      <React.Fragment>
        <IconButton
          edge="end"
          aria-label="menu"
          onClick={() => this.toggleDrawer("right", true)}
        >
          <Menu />
        </IconButton>
      </React.Fragment>
    );
  }
}

export default MobileNavigationBar;
