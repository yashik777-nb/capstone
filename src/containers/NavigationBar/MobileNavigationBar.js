import React from "react";
import { NavLink } from "react-router-dom";
import {
  IconButton,
  List,
  ListItem,
  ListItemText,
  Drawer,
} from "@material-ui/core";
import { withStyles } from "@material-ui/styles";
import { Menu } from "@material-ui/icons";

const styles = (theme) => ({
  list: {
    width: 250,
  },
  linkText: {
    textDecoration: `none`,
    textTransform: `uppercase`,
    color: `black`,
  },
});

class MobileNavigationBar extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      right: false,
    };
  }

  toggleDrawer = (anchor, open) => {
    this.setState({ [anchor]: open });
  };

  sideDrawerList = (anchor) => (
    <div
      className={this.props.list}
      role="presentation"
      onClick={() => this.toggleDrawer(anchor, false)}
      onKeyDown={() => this.toggleDrawer(anchor, false)}
    >
      <List component="nav">
        {this.props.navLinks.map(({ title, path }) => (
          <NavLink to={path} key={title} className={this.props.linkText}>
            <ListItem button>
              <ListItemText primary={title} />
            </ListItem>
          </NavLink>
        ))}
      </List>
    </div>
  );

  render() {
    return (
      <React.Fragment>
        <IconButton
          edge="end"
          aria-label="menu"
          onClick={() => this.toggleDrawer("right", true)}
        >
          <Menu fontSize="large" style={{ color: `white` }} />
        </IconButton>
        <Drawer
          anchor="right"
          open={this.state.right}
          onClose={() => this.toggleDrawer("right", false)}
        >
          {this.sideDrawerList("right")}
        </Drawer>
      </React.Fragment>
    );
  }
}

export default withStyles(styles, { withTheme: true })(MobileNavigationBar);
