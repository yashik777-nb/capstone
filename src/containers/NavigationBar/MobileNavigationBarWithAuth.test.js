import React from "react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import { NavLink } from "react-router-dom";
import { Drawer } from "@material-ui/core";
import { configure } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import { createMount } from "@material-ui/core/test-utils";
import { createMuiTheme, MuiThemeProvider } from "@material-ui/core/styles";
import configureStore from "redux-mock-store";
import toJson from "enzyme-to-json";

import MobileNavigationBar from "./MobileNavigationBar";

configure({ adapter: new Adapter() });

const navLinks = [
  { title: `About`, path: `/about`, toolTip: `About Application` },
  { title: `Issues`, path: `/issues`, toolTip: `Issues in the Applicaiton` },
  { title: `SignIn`, path: `/signin`, toolTip: `Sign In to Applciaiton` },
  { title: `Register`, path: `/register`, toolTip: `Register Yourself` },
];

const styles = {
  list: "",
  linkText: "",
};

const store = {
  users: {
    authenticated: true,
  },
};

const props = {
  classes: styles,
  authenticated: true,
  navLinks: navLinks,
};

const mockStore = configureStore();

describe("Mobile Navigation Bar Tests", () => {
  let wrapper;

  beforeEach(() => {
    const render = createMount();
    wrapper = render(
      <Provider store={mockStore(store)}>
        <BrowserRouter>
          <MobileNavigationBar {...props} />
        </BrowserRouter>
      </Provider>
    );
  });

  it("Should render seven NavLinks if authenticated for Mobile", () => {
    expect(wrapper.find(Drawer).length).toEqual(1);
  });

  test("Should Render Mobile navigation Bar", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
