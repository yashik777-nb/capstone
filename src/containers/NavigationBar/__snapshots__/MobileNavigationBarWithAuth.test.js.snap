// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mobile Navigation Bar Tests Should Render Mobile navigation Bar 1`] = `ReactWrapper {}`;
